tool/src/org/antlr/v4/codegen/model/RuleActionFunction.java

```diff
@@ -12,17 +12,19 @@
 import java.util.LinkedHashMap;
 
 public class RuleActionFunction extends OutputModelObject {
-	public String name;
-	public String ctxType;
-	public int ruleIndex;
+	public final String originalName;
+	public final String name;
+	public final String ctxType;
+	public final int ruleIndex;
 
 	/** Map actionIndex to Action */
 	@ModelElement public LinkedHashMap<Integer, Action> actions =
 		new LinkedHashMap<Integer, Action>();
 
 	public RuleActionFunction(OutputModelFactory factory, Rule r, String ctxType) {
 		super(factory);
-		name = r.name;
+		originalName = r.name;
```

Comment:
yep, this definitely seems backwards...

Reply:
Actually, there is a logic here:

* If an entity is located in tool package (analyzer) then it has `name` for original name and `escapedName` for runtime name (if presented)
* If an entity is located in codegen package then it has `name` for runtime name and `originalName` for grammar name.

It's for minimizing code changes.

Reply:
Well, things get blurred when we're looking at the tool and the Java runtime together so I think we need to make these consistent. name should always be the original.  if we have the potential for an escaped value, that field should be so named.